libsolidity/analysis/ImmutableValidator.h

```diff
@@ -28,8 +28,9 @@ namespace solidity::frontend
 
 /**
  * Validates access and initialization of immutable variables:
- * must be directly initialized in their respective c'tor
- * can not be read by any function/modifier called by the c'tor (or the c'tor itself)
+ * must be directly initialized in their respective c'tor or inline
+ * cannot be read before being initialized
+ * cannot be initialized using another immutable variable
```

Comment:
The second line is only true for state variables. Maybe reformulate as
```suggestion
 * cannot be read before being initialized
 * cannot be read when initializing state variables inline
```